src/Language/PureScript/DCE/CoreFn.hs

```diff
@@ -111,24 +111,39 @@ runDeadCodeElimination entryPoints modules = uncurry runModuleDeadCodeEliminatio
 
       -- | Find dependencies of an expression.
       deps :: Expr Ann -> [Key]
-      deps = go
+      deps = onExpr'
         where
-          (_, go, _, _) = everythingOnValues (++)
-            (const [])
-            onExpr
-            onBinder
-            (const [])
-
           -- | Build graph from qualified identifiers.
           onExpr :: Expr Ann -> [Key]
           onExpr (Var _ i) = [i | isQualified i]
           onExpr _ = []
 
-          onBinder :: Binder Ann -> [Key]
-          onBinder (ConstructorBinder _ _ c _) =
-            [fmap (Ident . runProperName) c]
+          onExpr' :: Expr Ann -> [Key]
+          onExpr' v@(Literal _ l) = foldl (++) (onExpr v) (map onExpr' (extractLiteral l))
+          onExpr' v@(Accessor _ _ e1) = onExpr v ++ onExpr' e1
+          onExpr' v@(ObjectUpdate _ obj vs) = foldl (++) (onExpr v ++ onExpr' obj) (map (onExpr' . snd) vs)
+          onExpr' v@(Abs _ _ e1) = onExpr v ++ onExpr' e1
+          onExpr' v@(App _ e1 e2) = onExpr v ++ onExpr' e1 ++ onExpr' e2
+          onExpr' v@(Case _ vs alts) = foldl (++) (foldl (++) (onExpr v) (map onExpr' vs)) (map onCaseAlternative alts)
+          onExpr' v@(Let _ ds e1) = foldl (++) (onExpr v) (map onBind ds) ++ onExpr' e1
+          onExpr' v = onExpr v
```

Comment:
could we use an explicit pattern match on remaining constructors?  This way once a new constructor will be added to `Expr` type, ghc will error, and we'll have a chance to make the right choice.